source/plugins/ruby/out_mdm.rb

```diff
@@ -102,6 +102,7 @@ def start
           else
             @http_client = Net::HTTP.new(@post_request_uri.host, @post_request_uri.port)
           end
+          @http_client.instance_eval { @ssl_context = OpenSSL::SSL::SSLContext.new(:TLSv1_2) }
```

Comment:
This (meaning negotiation) can fail if the metric endpoint enforces higher version of the TLS at some point, for example TLSv1_3.  I think, you have mentioned there are no server side logs,  but curios, you were able to get clues from client, what version of TLS version being used the client? probably, picking the higher supported TLS version can make both the client & server agree lower version so that we dont need to worry about  TLS enforcement at least for available TLS versions.

Reply:
These are the TLS versions supported by ingestion. Its literally tlsv1_2 alone. so what you do want to make agree by whom ?
![image](https://user-images.githubusercontent.com/10353076/118704843-632c6f80-b7cc-11eb-9da2-ab7972bc7f63.png).

Reply:
what I am saying, if ingestion endpoint makes TLS1.3 as supported version, then this code will break because the server wants to TLS 1.3 where as client hard coded for TLS1.2.

Reply:
yes, thats eactly what i said in the scrum yesterday. There is no negotiation is happening. we are forcing it to use v12

Reply:
negotiation still happens but client saying it can only support TLS12 or lower.  https://datatracker.ietf.org/doc/html/rfc5246#appendix-E.1
https://datatracker.ietf.org/doc/html/rfc5246#section-7.4.1.2 
If the server upgrades to new version and then connection closed . To be future proof, if we set TLS1.3 as client version, then it should work future & all existing scenarios.

Reply:
currently tls 1.3 is not supported by ingestion. see pic above.

Reply:
Also, how do we know they will go to tls13 and not sslv3?

Reply:
because tls13 will be new version and they cant go back to deprecated sslv3 version.

Reply:
you are predicting the future. they can go to a totally different ssl version like 14 when it becomes available.